frontend/src/pages/modelServing/screens/projects/useIsNIMAvailable.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect, useState } from 'react';
+import { SupportedArea, useIsAreaAvailable } from '~/concepts/areas';
+import { isNIMServingRuntimeTemplateAvailable } from '~/pages/modelServing/screens/projects/nimUtils';
+
+export const useIsNIMAvailable = (): boolean => {
+  const [isNIMAvailable, setIsNIMAvailable] = useState<boolean>(false);
+  const isNIMModelServingAvailable = useIsAreaAvailable(SupportedArea.NIM_MODEL).status;
+
+  useEffect(() => {
+    const checkNIMServingRuntime = async () => {
+      const isNIMRuntimeAvailable = await isNIMServingRuntimeTemplateAvailable();
+      setIsNIMAvailable(isNIMModelServingAvailable && isNIMRuntimeAvailable);
+    };
+
+    checkNIMServingRuntime();
+  }, [isNIMModelServingAvailable]);
+
+  return isNIMAvailable;
```

Comment:
Nit: Arguably you can just move the `isNIMModelServingAvailable` check to here -- and not add it to the dependency of the effect.

Reply:
Should we remove the dependency array completely? Leaving it empty causes an error, and if we remove it altogether, the effect will run after every render.

Reply:
`isNIMRuntimeAvailable` is not dependent on `isNIMModelServingAvailable`. They are exclusive to each other.

Reply:
I think we are fine here -- it was a nit.